pinot-broker/src/main/java/org/apache/pinot/broker/routing/v2/instanceselector/ReplicaGroupInstanceSelector.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.broker.routing.v2.instanceselector;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.pinot.common.metrics.BrokerMetrics;
+import org.apache.pinot.common.utils.HashUtil;
+
+
+/**
+ * Instance selector for replica-group routing strategy.
+ * <p>The selection algorithm will always evenly distribute the traffic to all replicas of each segment, and will select
+ * the same replica id for all segments with the same number of replicas. The algorithm is very light-weight and will
```

Comment:
This might be a dumb question: As per the existing design, don't we have a table level constraint that each segment will have the same number of replicas?

Reply:
Yes, but no guarantee on the number of replicas up in external view